panel/tests/test_server.py

```diff
@@ -23,12 +24,28 @@
 from panel.param import ParamFunction
 from panel.reactive import ReactiveHTML
 from panel.template import BootstrapTemplate
-from panel.tests.util import serve_and_request, serve_and_wait, wait_until
+from panel.tests.util import (
+    get_open_ports, serve_and_request, serve_and_wait, wait_until,
+)
 from panel.widgets import (
     Button, Tabulator, Terminal, TextInput,
 )
 
 
+@pytest.fixture(params=["tornado", "fastapi"])
+def server_implementation(request):
+    try:
+        importlib.import_module(request.param)
+    except Exception:
+        pytest.skip(f'{request.param!r} is not installed')
+    old = serve_and_wait.server_implementation
+    serve_and_wait.server_implementation = request.param
+    try:
+        yield
```

Comment:
```suggestion
        yield request.param
```

Mostly for convenience.